tracing/trace.go

```diff
@@ -159,7 +181,10 @@ func (t *trace) publish(ctx context.Context) error {
 	if !t.shouldSample() || t.tracer == nil {
 		return nil
 	}
-	return t.tracer.Record(ctx, t.toZipkinSpan())
+	if tracer := t.getTracer(); tracer != nil {
```

Comment:
This means if you override the global tracer with one that's not a `*Tracer`, no spans are ever published.

Reply:
aren't we actually trying to override it with a different tracer implementation? isn't it going to break a lot of things?

can you provide a high level explanation on how the library would work with the new tracer implementation?

Reply:
This is actually the behavior we desire. If you provide a tracer that is not a `*Tracer`, then `tracer.Record` is never called. However, presumably whatever tracer you do provide would record its spans and export them. Applying these changes makes this library compatible with other opentracing tracers, such as the opentracing bridge for opentelemetry, which will not be a `*Tracer` but would likely record and ship spans using an OTLP trace exporter.